automapper/src/main/java/com/valqueries/automapper/SqlDescriber.java

```diff
@@ -26,15 +23,14 @@ public SqlDescriber(DialectFactory dialectFactory) {
 		this.dialectFactory = dialectFactory;
 	}
 
-	public DbTable describe(TypeDescriber<?> typeDescriber, String tablename, Database database) {
+	public DbTable describe(TableToken tablename, Database database) {
 		try (IOrm tx = database.getOrm()) {
 			SqlDialect dialect = dialectFactory.get(database);
 			DbTable table = new DbTable();
-			table.columns.putAll(tx.query(dialect.describe(tablename), dialect::getDbRow).stream().collect(Collectors.toMap(DbRow::getField, Function.identity())));
-			table.index.putAll(tx.query(dialect.describeIndex(tablename), dialect::getDbIndex).stream().collect(Collectors.toMap(DbIndex::getKeyName, Function.identity(), (idx1, idx2) -> {
+			table.columns.putAll(tx.query(dialect.describe(tablename), dialect::getDescribeDbResult).stream().collect(Collectors.toMap(DbRow::getField, Function.identity())));
+			table.index.putAll(tx.query(dialect.describeIndex(tablename), dialect::getDescribeIndexResult).stream().collect(Collectors.toMap(DbIndex::getKeyName, Function.identity(), (idx1, idx2) -> {
 				if (idx1.getKeyName().equals(idx2.getKeyName())) {
-					idx2.getColumns().putAll(idx1.getColumns());
-					return null;
+					idx1.getColumns().addAll(idx2.getColumns());
```

Comment:
As we talk there should be more investigation on this lines. Maybe we can put a test here to check this!